### PR TITLE
Prompt user to select batch numbers

### PR DESCRIPTION
**Pull request by Alchez**

### CODE REVIEW

**Comment by vjFaLk** [comment]
Add screenshot please.

**Comment by Alchez** [comment]
@vjFaLk 

batch-select

File: erpnext/accounts/doctype/sales_invoice/sales_invoice.py [hunk 1]
```diff
@@ -107,7 +107,8 @@ def validate(self):
 		if not self.is_opening:
 			self.is_opening = 'No'
 
-		if self._action != 'submit' and self.update_stock and not self.is_return:
+		if self._action != 'submit' and self.update_stock and not self.is_return \
+			and frappe.db.get_single_value("Stock Settings", "automatically_set_batch_nos_based_on_fifo"):
```
Comment (vjFaLk) [inline erpnext/accounts/doctype/sales_invoice/sales_invoice.py:111]: Er, technically according to syntax, but not really?

File: erpnext/stock/doctype/delivery_note/delivery_note.py [hunk 2]
```diff
@@ -116,7 +116,8 @@ def validate(self):
 		self.validate_uom_is_integer("uom", "qty")
 		self.validate_with_previous_doc()
 
-		if self._action != 'submit' and not self.is_return:
+		if self._action != 'submit' and not self.is_return \
+			and frappe.db.get_single_value("Stock Settings", "automatically_set_batch_nos_based_on_fifo"):
```
Comment (vjFaLk) [inline erpnext/stock/doctype/delivery_note/delivery_note.py:120]: Er, technically according to syntax, but not really?

**Review by vjFaLk — CHANGES_REQUESTED** [review]

**Comment by vjFaLk** [comment]
Make this to Upstream, probably to develop.

**Comment by Alchez** [comment]
Made upstream: https://github.com/frappe/erpnext/pull/16353

**Comment by vjFaLk** [comment]
Thought - Maybe have a button to bring up that dialog again. In case the User accidentally dismisses the popup (which they do a lot) they can bring it back up. 

I know hitting "Save" will bring it up, but that's bad UX

**Comment by Alchez** [comment]
Yeah, good call.

**Review by vjFaLk — APPROVED** [review]

**Pull request merged**